### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by thiagomoraesp** [essence/essence#125]
Hi guys, i need to generate multiples youtube embeds in one page(about 50 videos), and the performance is not good at all, there's a way of increase performance?

**Comment by felixgirault** [comment]
Hi,

Essence is really about getting the most information possible on a media. 
It does make a request for each one, so it must indeed be quite awful with 50 videos!
But I don't know if we can do anything about cases like this in the lib itself...

What do you mean exactly by "generate embeds"? Do you call `$Essence->replace()` on a text?

And most importantly, do you need any of the information extracted by Essence, or do you just need the HTML code to embed the video?

**Comment by thiagomoraesp** [comment]
Im extracting with this method:

http://pastebin.com/M67GEXfV{

Extracting the html of the elements.
there are another way to call it?

**Comment by felixgirault** [comment]
There is no other way, but I was thinking that if you only need the HTML, you could maybe build it without essence.

Do you know where the URLs you are extracting come from ?
If there are all from a single source, it could be really simple.

**Comment by thiagomoraesp** [comment]
Another thing that i was thinking about is make a "cache" using essence, but saving the embed code on the database to do not need to consult the providers on every query.

*sorry my bad english

**Comment by thiagomoraesp** [comment]
I don't know, so i need essence.

**Comment by felixgirault** [comment]
A cache could be a good first step, and it is not too hard to implement.
If the code you showed me runs only once and saves everything in database, your problem will probably be gone.

**Comment by felixgirault** [comment]
Thinking about it, it would be nice to have an "offline" mode, in which essence could get minimal information about the URL, like HTML code and title.

It would require some time to fit this nicely into the lib, so I added an issue to keep that in mind :) https://github.com/essence/essence/issues/126

**Comment by thiagomoraesp** [comment]
thanks felix =)
